core/src/main/java/net/sourceforge/jnlp/util/UrlWhiteListUtils.java

```diff
@@ -0,0 +1,114 @@
+
+public class UrlWhiteListUtils {
+    private final static Logger LOG = LoggerFactory.getLogger(UrlWhiteListUtils.class);
+    public static final String HTTPS = "https";
+    public static final String HTTP = "http";
+    public static final String HTTP_PORT = "80";
+    public static final String HTTPS_PORT = "443";
+
+    public static boolean isUrlProtocolEqual(URL url1, URL url2) {
+        Assert.requireNonNull(url1, "url");
+        Assert.requireNonNull(url2, "url");
+        return url1.getProtocol().equals(url2.getProtocol());
+    }
+
+    public static boolean isUrlHostWithWildcardEqual(URL urlWithWildcard, URL url) {
+        Assert.requireNonNull(urlWithWildcard, "url");
+        Assert.requireNonNull(url, "url");
+        String[] wlUrlHostParts = urlWithWildcard.getHost().split("\\.");
+        String[] urlHostParts = url.getHost().split("\\.");
+        // proto://*:port
+        if (wlUrlHostParts.length == 1 && wlUrlHostParts[0].length() ==1 && wlUrlHostParts[0].charAt(0) == '*') {
+            return true;
+        }
+        if (wlUrlHostParts.length != urlHostParts.length) {
+            return false;
+        }
+        boolean result = true;
+        for (int i = 0; i < wlUrlHostParts.length; i++) {
+            result = result && (wlUrlHostParts[i].charAt(0) == '*' || wlUrlHostParts[i].equals(urlHostParts[i]));
+        }
+        return result;
+    }
+
+    public static boolean isUrlInWhitelist(final URL url, final List<String> whiteList, final boolean allowLocalhost, final boolean exactMatch) {
+        Assert.requireNonNull(url, "url");
+        Assert.requireNonNull(whiteList, "whiteList");
+
+        if (whiteList.isEmpty()) {
+            return true; // empty whitelist == allow all connections
+        }
+
+        // if host is null or "" or it is localhost or loopback
+        if (allowLocalhost && IpUtil.isLocalhostOrLoopback(url)) {
+            return true; // local server need not be in whitelist
+        }
+
+        boolean result = whiteList.stream().anyMatch(wlUrlStr -> {
+                try {
+                    if (exactMatch) {
+                        final URL wlUrl = new URL(wlUrlStr);
+                        return UrlUtils.notNullUrlEquals(url, wlUrl);
+                    } else {
+                        final URL expandedWlUrl = new URL(UrlWhiteListUtils.expandedWhiteListUrlString(wlUrlStr));
+                        final boolean isUrlProtocolEqual = isUrlProtocolEqual(expandedWlUrl, url);
+                        final boolean isUrlHostWithWildcardEqual = isUrlHostWithWildcardEqual(expandedWlUrl, url);
+                        final boolean isUrlPortEqual = expandedWlUrl.getPort() != -1 ? expandedWlUrl.getPort() == url.getPort() : true;
+                        return isUrlProtocolEqual && isUrlHostWithWildcardEqual && isUrlPortEqual;
+                    }
+                } catch (Exception e) {
+                    LOG.debug("Bad white list url: " + wlUrlStr);
```

Comment:
this is much more than a debug when the white list contains something invalid. Should be warning and log message should contain the information that this URL will be ignored.

Reply:
+1

Reply:
As we catch the exception here and just log we do not have the chance to show a warning decorator icon next to the whitelist in Settings UI with a tooltip "Your whitelist contains malformed entries" or similar to inform the user that something is not perfect with his configuration. Alternatively we could allow comments in white lists and whenever we found an invalid one, we insert it like
#Invalid Entry: <invalid entry>
that makes the problem visible to the user